src/lib/mina/zkapp.ts

```diff
@@ -1058,21 +1059,18 @@ super.init();
       chainStatus: string;
     }[]
   > {
+    const queryFilterOptions: EventActionFilterOptions = {};
+    if(start.greaterThan(UInt32.from(0)).toBoolean()) {
+      queryFilterOptions.from = start;
+    }
+    if(end) {
+      queryFilterOptions.to = end;
+    }
     // filters all elements so that they are within the given range
     // only returns { type: "", event: [] } in a flat format
     let events = (
-      await Mina.fetchEvents(this.address, this.self.body.tokenId, {
-        from: start,
-        to: end,
-      })
+      await Mina.fetchEvents(this.address, this.self.body.tokenId, queryFilterOptions)
     )
-      .filter((eventData) => {
```

Comment:
Removing this filter because it is a duplicate of the filter applied by the archive node.  Was it intentional to leave this as insurance?

Reply:
yeah, this implementation was written when the archive node API was still in an early WIP phase and simply always returned the full list of events

Reply:
so, good to remove